### PR TITLE
Make Python 2 an optional dependency for Bazel build

### PR DESCRIPTION
**Pull request by gnossen**
This PR makes a Python 2 runtime with headers an optional dependency for Bazel builds. In case no proper Python 2 environment is detected, we create a rule like

```Starlark
filegroup(
  name="_python2_include",
  srcs=[],
)
```

This empty `filegroup` rule passes the analysis phase, but will fail during the build phase of any Python 2 targets, which are all test targets within our repo, and which downstream repos will not be touching.

This will also be backported to 1.33.

### CODE REVIEW

File: third_party/py/python_configure.bzl [hunk 1]
```diff
@@ -279,19 +286,29 @@ def _create_single_version_package(
...
+
+    # A for loop is used in place of a (non-existent in Starlark) goto.
+    for _ in [None]:
+        python_bin = _get_python_bin(repository_ctx, bin_path_key, default_bin_path, allow_absent)
+        if python_bin == None and allow_absent:
+            python_include_rule = empty_include_rule
+            break
+        if _check_python_bin(repository_ctx, python_bin, bin_path_key, allow_absent) == None and allow_absent:
+            python_include_rule = empty_include_rule
+            break
```
Comment (lidizheng) [inline third_party/py/python_configure.bzl:302]: Optional: Could save a few lines
```python
if python_bin == None or _check_python_bin(...):
    if allow_absent:
        python_include_rule = "filegroup(\n  name=\"{}_include\",\n  srcs=[],\n)".format(variety_name)
        break
```

**Review by lidizheng — APPROVED** [review]
LGTM! Are users asking for this feature?

File: third_party/py/python_configure.bzl (hunk 1 above)
Comment (gnossen) [reply third_party/py/python_configure.bzl]: Good suggestion. Much cleaner.

**Comment by gnossen** [comment]
Yes, the users I mentioned this morning were having trouble getting Python 2 installed in build environments that otherwise didn't need it.

**Comment by gnossen** [comment]
Known failures:
 - Pip connection error
 - Network error downloading Python 3.8

We should probably add some retries in there.

**Comment by ChrisCummins** [comment]
Hi, did this land in v1.34.0? I am still getting python_configure errors building in an `ubuntu:18.04` docker image using bazel. The workaround is to install `python-dev`.

Cheers,
Chris

**Pull request merged**